### PR TITLE
Enable compile-time type ID checks in dev mode

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This change ensures that all types that get added to a `caf::message` have a valid type ID assigned to them, and makes the necessary changes for this to work.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file. I recommend starting with `CMakeLists.txt` and `vast/fwd.hpp`.

### CODE REVIEW

File: libvast/vast/detail/stable_map.hpp [hunk 1]
```diff
@@ -34,10 +36,4 @@ struct stable_map_policy {
   }
 };
 
-/// A map abstraction over an unsorted `std::vector`.
-template <class Key, class T,
-          class Allocator = std::allocator<std::pair<Key, T>>>
-using stable_map = vector_map<Key, T, Allocator, stable_map_policy>;
```
Comment (tobim) [inline libvast/vast/detail/stable_map.hpp:40]: Here should be a comment that directs the reader to `vast/fwd.hpp`.

**Review by tobim — CHANGES_REQUESTED** [review]
Looks great! Just one nit from me.

File: libvast/vast/detail/stable_map.hpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/vast/detail/stable_map.hpp:40]: Done.

**Review by tobim — APPROVED** [review]

File: libvast/vast/fwd.hpp [hunk 2]
```diff
@@ -200,6 +203,20 @@ struct v0;
...
 } // namespace fbs
 
+namespace detail {
+
+struct stable_map_policy;
+
+template <class, class, class, class>
+class vector_map;
+
+/// A map abstraction over an unsorted `std::vector`.
```
Comment (mavam) [inline libvast/vast/fwd.hpp:213]:
```suggestion
```

Comment (dominiklohmann) [reply libvast/vast/fwd.hpp:213]: This is an alias definition with a defaulted template argument and we can only have it once, so I moved it over with its documentation.

**Comment by dominiklohmann** [comment]
Merging this despite the failed CI, the error is the usual node queries unit test issue.

**Pull request merged**